admin/admin-filters-media.php

```diff
@@ -51,23 +51,34 @@ public function __construct( &$polylang ) {
 	 */
 	public function attachment_fields_to_edit( $fields, $post ) {
 		if ( 'post.php' === $GLOBALS['pagenow'] ) {
-			return $fields; // Don't add anything on edit media panel for WP 3.5+ since we have the metabox.
+			return $fields; // Don't add anything on edit media panel since we have the metabox.
 		}
 
 		$post_id = $post->ID;
-		$lang = $this->model->post->get_language( $post_id );
+		$lang    = $this->model->post->get_language( $post_id );
+		$user    = new User();
+
+		if ( ( empty( $lang ) && $user->is_translator() ) || ( ! empty( $lang ) && ! $user->can_translate( $lang ) ) ) {
+			// The user is not allowed to edit this attachment.
+			$languages = $this->model->languages->get_list();
+			$disabled  = true;
+		} else {
+			$languages = $this->model->languages->filter( 'translator' )->get_list();
+			$disabled  = false;
+		}
```

Comment:
For the languages list, we don't need a complete list if the select is disabled.
For the test, I find it clearer like that (see also my comment below):
```suggestion
		$is_allowed = empty( $lang ) ? ! $user->is_translator() : $user->can_translate( $lang );
		$languages  = $is_allowed ? $this->model->languages->filter( 'translator' )->get_list() : $lang;
```